frontend/src/reducer/notificationPage.js

```diff
@@ -74,16 +79,46 @@ function getMainContentId (notification) {
     : notification.content.id
 }
 
-// FIXME - GB - 2022-04-21 - this code is very similar to activityDisplayFilter
-// in withActivity and ActivityList, and can be refactor
-// See https://github.com/tracim/tracim/issues/4677
-function notificationListDisplayFilter (notificationList, spaceList, unreadNotificationCount) {
+/**
+ * FIXME - GB - 2022-04-21 - This code is very similar to activityDisplayFilter
+ * in withActivity and ActivityList, and can be refactor
+ * See https://github.com/tracim/tracim/issues/4677
+ * FIXME - MP - 2022-10-18 - This function allows us to filter when loading more notifications.
+ * However there is a similar loginc in ReduxTlmDispatcher.js when we receive a TLM.
+ * https://github.com/tracim/tracim/issues/5946
+ *
+ * Filter the notification list
+ * @param {int} userId
+ * @param {*} notificationList
+ * @param {*} spaceList
+ * @param {*} unreadNotificationCount
+ * @returns Notification list filtered and the unread notification count
+ */
+function notificationListDisplayFilter (
+  userId, notificationList, spaceList, unreadNotificationCount
+) {
   let newUnreadNotificationCount = unreadNotificationCount
   const newNotificationList = notificationList.map((notification) => {
-    const [entityType] = notification.type.split('.')
+    const [entityType, , subTpe] = notification.type.split('.')
+
+    // NOTE - MP - 2022-10-18 - Since we are filtering twice, the -1 verification is if we are
+    // doing this function in the `addNotification` case, which in this case is already filtered
+    if (userId !== -1) {
+      if (entityType === TLM_ET.CONTENT && subTpe === TLM_ST.TODO) {
+        if (notification.content.assignee.username) {
+          if (notification.content.assignee.userId !== userId) {
+            if (!notification.read) newUnreadNotificationCount--
+            return null
```

Comment:
Although it works, the map with null returns and a filter afterwards seems a bit difficult to read (besides making two loops in the array). I wonder if a forEach that builds the array properly or just the filter (returning true and false instead of null and notification) would be better.
I saw that you just used the logic that was already there before, so no need to change, the comment is just a reflection.

Reply:
No problem